protocol/contracts/templegold/TempleGoldStaking.sol

```diff
@@ -0,0 +1,521 @@
+        uint256 stakerBalance = _balances[staker];
+        _withdrawFor(staker, msg.sender, stakerBalance, true, staker);
+        return stakerBalance;
+    }
+    
+    /**
+     * @notice Distributed TGLD rewards minted to this contract to stakers
+     * @dev This starts another epoch of rewards distribution. Calculates new `rewardRate` from any left over rewards up until now
+     */
+    function distributeRewards() external override whenNotPaused updateReward(address(0)) {
+        if (msg.sender != distributionStarter) { revert CommonEventsAndErrors.InvalidAccess(); }
+        if (totalSupply == 0) { revert NoStaker(); }
+        // Mint and distribute TGLD if no cooldown set
+        if (lastRewardNotificationTimestamp + rewardDistributionCoolDown > block.timestamp) 
+                { revert CannotDistribute(); }
+        _distributeGold();
+        uint256 rewardAmount = nextRewardAmount;
+        // revert if next reward is 0 or less than reward duration (final dust amounts)
+        if (rewardAmount < rewardDuration ) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        nextRewardAmount = 0;
+        _notifyReward(rewardAmount);
+        lastRewardNotificationTimestamp = uint32(block.timestamp);
+    }
+
+    /**
+     * @notice Gets the current votes balance for `account`
+     * @param account The address to get votes balance
+     * @return The number of current votes for `account`
+     */
+    function getCurrentVotes(address account) external override view returns (uint256) {
+        uint256 nCheckpoints = numCheckpoints[account];
+        return nCheckpoints > 0 ? _checkpoints[account][nCheckpoints - 1].votes : 0;
+    }
+
+    /**
+     * @notice Determine the prior number of votes for an account as of a block number
+     * @dev Block number must be a finalized block or else this function will revert to prevent misinformation.
+     * @param account The address of the account to check
+     * @param blockNumber The block number to get the vote balance at
+     * @return The number of votes the account had as of the given block
+     */
+    function getPriorVotes(address account, uint256 blockNumber) external override view returns (uint256) {
+        if (blockNumber >= block.number) { revert InvalidBlockNumber(); }
+
+        uint256 nCheckpoints = numCheckpoints[account];
+        if (nCheckpoints == 0) {
+            return 0;
+        }
+
+        // First check most recent balance
+        if (_checkpoints[account][nCheckpoints - 1].fromBlock <= blockNumber) {
+            return _checkpoints[account][nCheckpoints - 1].votes;
+        }
+
+        // Next check implicit zero balance
+        if (_checkpoints[account][0].fromBlock > blockNumber) {
+            return 0;
+        }
+
+        uint256 lower = 0;
+        uint256 upper = nCheckpoints - 1;
+        while (upper > lower) {
+            uint256 center = upper - (upper - lower) / 2; // ceil, avoiding overflow
```

Comment:
I understand that the comment refers to the division `(upper - lower) / 2`. 

The result is a **floor** division, which will ensure the subtraction `upper - ...` will never overflow. 
```suggestion
            uint256 center = upper - (upper - lower) / 2; // floor, avoiding overflow
```